### PR TITLE
Add totals bar to bar graph

### LINKED ISSUE #136: Add a "total" site with total counts across all sites
Scope:

* Add a bar to each chart that has total values across all of the sites.

### PR DESCRIPTION
**Pull request by Elkrival**
Total counts bar displayed on bar graph.

Screen Shot 2022-08-19 at 4 12 37 PM

### CODE REVIEW

File: server/controllers/chartsController.js [hunk 1]
```diff
@@ -2,6 +2,8 @@ import { ObjectID } from 'mongodb'
 
 import { collections } from '../utils/mongoCollections'
 
+const Totals = 'Totals'
```
Comment (mikestone14) [inline server/controllers/chartsController.js]: can we be a little more descriptive here and also use all caps for the constant?

```suggestion
const TOTALS_STUDY = 'Totals'
```

File: views/components/Graphs/BarGraph.jsx [hunk 2]
```diff
@@ -23,7 +23,8 @@ const BarGraph = ({ graph }) => {
         <VictoryVoronoiContainer
           labels={({ datum: { study, studyTarget, count, valueLabel } }) => {
             const showToolTip = study && count
-            return valueLabel !== 'N/A' && showToolTip
+            const notTotals = study !== 'Totals'
+            return valueLabel !== 'N/A' && showToolTip && notTotals
```
Comment (mikestone14) [inline views/components/Graphs/BarGraph.jsx]: I wonder if we want to update the `showToolTip` logic with this change. What do you think? Also, do you think we could leverage the same constant in both places for the totals study?

```js
const showToolTip = study && count && study !== 'Totals'
```

Comment (Elkrival) [reply views/components/Graphs/BarGraph.jsx]: https://github.com/AMP-SCZ/dpdash/pull/137/commits/63f1716f2a06552e8dee7d012e9a7733b206fbaa

File: server/controllers/chartsController.js (hunk 1 above)
Comment (Elkrival) [reply server/controllers/chartsController.js]: https://github.com/AMP-SCZ/dpdash/pull/137/commits/f87a63ed70a41555dabd1aeb8becb444ca08a66c

File: constants/constants.js [hunk 3]
```diff
@@ -0,0 +1 @@
+export const TOTALS_STUDY = 'Totals'
```
Comment (mikestone14) [inline constants/constants.js]: we can put this file in `constants/index.js` so we don't need to import it with the repeating `constants` directory and file.

```js
import { TOTALS_STUDY } from '../../constants/constants'
```

vs

```js
import { TOTALS_STUDY } from '../../constants'
```

File: server/controllers/chartsController.js [hunk 4]
```diff
@@ -48,13 +49,26 @@ const postProcessData = (data, studyTotals) => {
...
   processedData['N/A'] = notAvailableArray
-
+  Object.keys(processedData).forEach((key) => {
+    processedData[key].sort(function (studyA, studyB) {
+      return studyA.study === TOTALS_STUDY
+        ? -1
+        : studyB.study === TOTALS_STUDY
+        ? 1
+        : 0
+    })
```
Comment (mikestone14) [inline server/controllers/chartsController.js]: can we remove the nested ternary? Those are pretty tough to follow.

**Review by mikestone14 — APPROVED** [review]
Looks good after the comments are addressed 🚢

File: server/controllers/chartsController.js (hunk 4 above)
Comment (Elkrival) [reply server/controllers/chartsController.js]: https://github.com/AMP-SCZ/dpdash/pull/137/commits/6c921e8f5dfccbd2c7666d80b951df5613819ba9

File: constants/constants.js (hunk 3 above)
Comment (Elkrival) [reply constants/constants.js]: https://github.com/AMP-SCZ/dpdash/pull/137/commits/6c921e8f5dfccbd2c7666d80b951df5613819ba9

**Pull request merged**